plugins/inputs/mesos/mesos.go

```diff
@@ -418,29 +437,63 @@ var client = &http.Client{
 	Timeout:   time.Duration(4 * time.Second),
 }
 
+func basicAuth(username, password string) string {
+	auth := username + ":" + password
+	return base64.URLEncoding.EncodeToString([]byte(auth))
+}
+
+func getEndpointParams(address string) map[string]string {
+
+	if strings.HasPrefix(address, "http") != true {
+		address = "http://" + address
+	}
+
+	u, err := url.Parse(address)
+	if err != nil {
+		panic(err)
+	}
+
+	host, port, err := net.SplitHostPort(u.Host)
+	if err != nil {
+		panic(err)
+	}
+
+	return map[string]string{
+		"host":   host,
+		"port":   port,
+		"scheme": u.Scheme,
+	}
+}
+
 // TaskStats struct for JSON API output /monitor/statistics
 type TaskStats struct {
 	ExecutorID  string                 `json:"executor_id"`
 	FrameworkID string                 `json:"framework_id"`
 	Statistics  map[string]interface{} `json:"statistics"`
 }
 
-func (m *Mesos) gatherSlaveTaskMetrics(address string, defaultPort string, acc telegraf.Accumulator) error {
+func (m *Mesos) gatherSlaveTaskMetrics(address, username, password string, acc telegraf.Accumulator) error {
 	var metrics []TaskStats
-
-	host, _, err := net.SplitHostPort(address)
-	if err != nil {
-		host = address
-		address = address + defaultPort
-	}
+	var endpoint map[string]string
+	endpoint = getEndpointParams(address)
 
 	tags := map[string]string{
-		"server": host,
+		"server": endpoint["host"],
 	}
 
 	ts := strconv.Itoa(m.Timeout) + "ms"
 
-	resp, err := client.Get("http://" + address + "/monitor/statistics?timeout=" + ts)
+	req, err := http.NewRequest(
+		"GET",
+		endpoint["scheme"]+"://"+endpoint["host"]+":"+endpoint["port"]+"/monitor/statistics?timeout="+ts,
+		nil,
+	)
+
+	if len(username) > 0 && len(password) > 0 {
+		req.Header.Add("Authorization", "Basic "+basicAuth(username, password))
```

Comment:
there is a Request.SetBasicAuth function that I think you should be using instead: https://golang.org/pkg/net/http/#Request.SetBasicAuth